src/examples/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,199 @@
+{
+    std::vector<UInt> normalized;
+    for (unsigned int i = 0; i < v.size(); ) {
+        unsigned int val = 0;
+        unsigned int j;
+        for (j = i; j < v.size() && j < i + cellsPerColumn; j++) {
+            val |= v[j];
+        }
+        normalized.push_back(val);
+        i = j;
+    }
+    return normalized;
+}
+
+// the chain of: input -> encoder -> spatial pooler -> temporal memory -> anomaly -> output
+int main()
+{
+    using namespace nupic;
+    using namespace nupic::algorithms::spatial_pooler;
+    using namespace nupic::algorithms::Cells4;
+    using namespace nupic::algorithms::anomaly;
+
+    std::string data("data.csv");
+    const UInt COLS = 2048;
+    const UInt CELLS = 4;
+
+    // Open the file or exit with error code 1
+    std::ifstream in(data.c_str());
+    if (!in.is_open()) return 1;
+
+    // Declare a vector for each column in the CSV file
+    std::vector<float> column1;
+
+    // Read the CSV file into a couple of vectors
+    std::string line;
+    while (getline(in, line)) {
+        // Use a boost tokenizer to parse a CSV line
+        boost::tokenizer<boost::escaped_list_separator<char>> tok(line);
+        std::vector<std::string> tokens(tok.begin(), tok.end());
+
+        // Skip line if it the aren't 2 elements at least.
+        if (std::distance(tok.begin(), tok.end()) < 1) continue;
+
+        // Store both items
+        column1.push_back(std::stof(tokens[0]));
+    }
+
+    std::cout << "Column1: ";
+    print_vector(column1, ' ');
+    std::cout << std::endl;
+
+    ScalarEncoder encoder{
+        25,   // w: Active bits per value
+        0,   // min value
+        10,  // max value
+        250,  // n: bits per output value,
+        0,
+        0,
+        true // Clip input to min/max vlues if needed.
+    };
+
+    // Spatial Pooler
+    SpatialPooler sp{
+        std::vector<UInt>{encoder.getOutputWidth()}, // Input dimensions = encoder width
+        std::vector<UInt>{COLS}  // Number of colums per dimension
+    };
+
+    // Temporal Pooler
+    Cells4 tp{
+        COLS,    // Number of columns
+        CELLS,     // Cells per column
+        12, 
+        8, 
+        15, 
+        5, 
+        .5, 
+        .8, 
+        1.0, 
+        .1, 
+        .1, 
+        0.0, 
+        false, 
+        42, 
+        true, 
+        false
+
+    };
+
+    std::vector<UInt> spOutput(COLS);
+    std::vector<UInt> lastTPOutput;
+
+    int iteration = 0;
+    for (auto& elem: column1) {
+        auto scalar = std::vector<Real32>(encoder.getOutputWidth());
+        encoder.encodeIntoArray(elem, scalar.data());
+
+        // For some reason, the ScalarEncoder outputs a Real32 vector,
+        // which is incomaptible with the SpatialPooler's input, hence a new 
+        // vector is created with the required type, and the elements
+        // are converted.
+        auto uint_scalar = vectorTypeCast<Real32, UInt>(scalar);
```

Comment:
this is what I was referring to.. 
// For some reason, the ScalarEncoder outputs a Real32 vector,
+        // which is incomaptible with the SpatialPooler's input, hence a new 
+        // vector is created with the required type, and the elements
+        // are converted.